ISISComputingGroup/ibex_gui
# Ticket2254: Fixed bugs around Devicescreens target properties

### Description of work

I moved the target properties in the device screens into a separate viewmodel. This made a number of bugs a bit more obvious to fix. See https://github.com/ISISComputingGroup/IBEX/issues/2254 for the specific bugs that this was specifically targeted towards.

### To test

* Ensure that properties of screens can be modified as expected
* Check that you can no longer reproduce the bugs outlined in the original ticket
* Check that the unit tests still all run correctly

---

#### Code Review

    - Manual system tests for the functionality should be added if there are no automated tests
    - Manual system tests can be removed from the template if they are covered by suitable automated tests

### Functional Tests


### Final Steps
